allas checks and updates (#2165)
CSCfi/csc-user-guide: docs/data/Allas/accessing_allas.md

## Accessing Allas from the web browser

At the moment CSC provides several web browser interfaces for Allas:

The **WWW interfaces of Puhti and Mahti** are connected to Allas. 
These interfaces allow you transfer files and directories between your local computer and Allas as well as
between CSC supercomputers and Allas.

> * [Instuctions for using Allas in Puhti and Mahti web interfaces](../../computing/webinterface/file-browser.md)

Reviewer: ```suggestion
* [Instructions for using Allas in Puhti and Mahti web interfaces](../../computing/webinterface/file-browser.md)
```

* [Puhti web interface](https://www.puhti.csc.fi)
* [Mahti web interface](https://www.mahti.csc.fi)

The OpenStack Horizon web interface in **cPouta** provides easy-to-use basic functions for data management in Allas. This interface can only be used for files smaller than 5 GB.

* [Web client – OpenStack Horizon Dashboard](./using_allas/web_client.md)
* [cPouta Web Interface](https://pouta.csc.fi)

**SD Connect** provides an interface for storing and sharing sensitive data. 

> This service is based of Allas but we don't recommend it for other than sensitive data.

Reviewer: ```suggestion
This service is based on Allas but we don't recommend it for other than sensitive data.
```

* [SD Connect instructions](../sensitive-data/sd_connect.md)
* [SD Connect interface](https://sd-connect.csc.fi)

## Accessing Allas in the CSC computing environment and other Linux platforms

> Puhti and Mahti servers at CSC support many different command line tools for using Allas, These include

Reviewer: ```suggestion
CSC supercomputers Puhti and Mahti support many different command line tools for using Allas, These include
```

Note that the tools listed above utilize two different protocols: _Swift_ and _S3_. Data uploaded using one protocol is not necessary compatible with another protocol. 

> The software listed above can be used other machines too like in a virtual machine running in cPouta or your own laptop. 

Reviewer: ```suggestion
The software listed above can also be used on other devices, for example a virtual machine running in cPouta or your own laptop.
```

In Puhti and Mahti the Allas tools listed above are installed by CSC and provided through _allas_ module.
In order to use Allas in Puhti or Mahti, first load the Allas module:
```text
module load allas
```
Allas access for a specific project using swift protocol can then be enabled:
```text
allas-conf
```
To enable S3 protocol, use option `-m S3`
```text
allas-conf -m S3
```

> The _allas-conf_ command prompts for your CSC password (the same that you use to login to CSC servers). It lists your Allas projects and asks you to define a project (if not already defined as an argument). _allas-conf_ generates an _rclone_ configuration file for the Allas service and authenticates the connection to the selected project. allas-conf enables you to use only one Allas project at a time in one session. The project you are using in Allas does not need to match the project you are using in Puhti or Mahti, and you can switch to another project by running _allas-conf_ again.

Reviewer: ```suggestion
The `allas-conf` command prompts for your CSC password (the same that you use to login to CSC servers). It lists your Allas projects and asks you to define a project (if not already defined as an argument). `allas-conf` generates an `rclone` configuration file for the Allas service and authenticates the connection to the selected project. `allas-conf` enables you to use only one Allas project at a time in one session. The project you are using in Allas does not need to match the project you are using in Puhti or Mahti, and you can switch to another project by running `allas-conf` again.
```

> In the case of swift-protocol, the authentication information is stored in the shell variables *OS_AUTH_TOKEN* and *OS_STORAGE_URL* and is valid for up to eight hours. However, you can refresh the authentication at any time my running _allas-conf_ again. The environment variables are available only for that login session, so if you start another shell session, you need to authenticate again in there to access Allas.

Reviewer: ```suggestion
In the case of the Swift protocol, the authentication information is stored in the `OS_AUTH_TOKEN` and `OS_STORAGE_URL` environment variables and is valid for up to eight hours. However, you can refresh the authentication at any time by running `allas-conf` again. The environment variables are set only for the current login session, so you need to configure authentication individually for each shell with which you wish to access Allas.
```

> In the case of S3 protocol, the authentication information is stored to configuration files in your home directory. Thus the authentication will be used for all sessions in in the machine in uses. The authentication does not have expiration time.

Reviewer: ```suggestion
In the case of the S3 protocol, the authentication information is stored in configuration files located in your home directory on the device. The same authentication is used for all login sessions and it does not have an expiration time.
```

> Once Allas connection is configured, you can start using Allas with the tools listed above. 

Reviewer: ```suggestion
Once an Allas connection is configured, you can start using the object storage with the tools listed above. 
```

For example following tools can be used:

* [Cyberduck](./using_allas/cyberduck.md) provides easy to use graphical interface for moving data between local computer and Allas.

> * [Rclone](./using_allas/rclone_local.md) is an command line tool that provides very effective way to use Allas from any operating system.

Reviewer: ```suggestion
* [Rclone](./using_allas/rclone_local.md) is a command line tool that provides a very effective way to use Allas on any operating system.
```

> * [a-tools](./using_allas/a_commands.md) these Allas specific commands can be installed in Mac OSX and Linux machines but not to Windows

Reviewer: ```suggestion
* [a-tools](./using_allas/a_commands.md) are Allas-specific commands that can be installed on macOS and Linux devices, but not ones running a Windows operating system.
```
